### PR TITLE
Fix RMS Object Construction

### PR DESCRIPTION
**Pull request by mjohnson541**
This fixes a few different bugs related to construction of RMS objects for more complex kinetics types particularly Multi/Pdep/Arrhenius and falloff types. 

Best test is to run RMG with libraries containing these kinetic types.

### CODE REVIEW

**Review by mazeau — COMMENTED** [review]
happy to approve! thanks so much, matt!

File: rmgpy/rmg/reactors.py [hunk 1]
```diff
@@ -412,23 +426,35 @@ def to_rms(obj, species_names=None, rms_species_list=None):
         coeffs = obj.coeffs.value_si.tolist()
         return rms.Chebyshev(coeffs, Tmin, Tmax, Pmin, Pmax)
     elif isinstance(obj, ThirdBody):
-        arr = to_rms(obj.arrheniusLow)
-        efficiencies = {spc.label: float(val) for spc, val in obj.efficiencies.items() if val != 1}
-        return rms.ThirdBody(arr, nameefficiencies=efficiencies)
+        arrstr = arrhenius_to_julia_string(obj.arrheniusLow)
+        efficiencies = {species_names[i] : float(val) for i, val in enumerate(obj.get_effective_collider_efficiencies(rmg_species)) if val != 1}
```
Comment (davidfarinajr) [inline rmgpy/rmg/reactors.py:430]: Are we sure that `species_names` is in the same order as `rmg_species`?  For a phase, it seems like they should be, not sure if we are also using this for interphase.

**Review by davidfarinajr — CHANGES_REQUESTED** [review]
Thanks for fixing this @mjohnson541 !  I found a couple of potential bugs in `add_reaction` Phase function (see comments).  Also, I am not 100% sure that species_names is in the same order as `rmg_species`, and I didn't follow the code all the way through to check, but it looks like they should be?

**Review by davidfarinajr — APPROVED** [review]

**Comment by davidfarinajr** [comment]
I just rebased and approved.  I'll merge it after the tests pass

**Comment by NUSCondy** [comment]
Hi,
I have updated to this main version and try to run and encountered this:
**Traceback (most recent call last):
  File "//home/zhuqiren/anaconda3/envs/rmg_env/lib/python3.7/site-packages/julia/pseudo_python_cli.py", line 308, in main
    python(**vars(ns))
  File "//home/zhuqiren/anaconda3/envs/rmg_env/lib/python3.7/site-packages/julia/pseudo_python_cli.py", line 59, in python
    scope = runpy.run_path(script, run_name="__main__")
  File "//home/zhuqiren/anaconda3/envs/rmg_env/lib/python3.7/runpy.py", line 263, in run_path
    pkg_name=pkg_name, script_name=fname)
  File "//home/zhuqiren/anaconda3/envs/rmg_env/lib/python3.7/runpy.py", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File "//home/zhuqiren/anaconda3/envs/rmg_env/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/zhuqiren/RMG-Py/rmg.py", line 48, in <module>
    from rmgpy.rmg.main import RMG, initialize_log, process_profile_stats, make_profile_graph
  File "/home/zhuqiren/RMG-Py/rmgpy/rmg/main.py", line 55, in <module>
    from rmgpy.rmg.model import Species, CoreEdgeReactionModel
  File "/home/zhuqiren/RMG-Py/rmgpy/rmg/model.py", line 41, in <module>
    import rmgpy.data.rmg
  File "/home/zhuqiren/RMG-Py/rmgpy/data/rmg.py", line 38, in <module>
    from rmgpy.data.base import ForbiddenStructures
  File "/home/zhuqiren/RMG-Py/rmgpy/data/base.py", line 44, in <module>
    from rmgpy.kinetics.uncertainties import RateUncertainty
  File "/home/zhuqiren/RMG-Py/rmgpy/kinetics/__init__.py", line 30, in <module>
    from rmgpy.kinetics.model import KineticsModel, PDepKineticsModel, TunnelingModel, \
  File "__init__.pxd", line 918, in init rmgpy.kinetics.model
ValueError: numpy.ufunc size changed, may indicate binary incompatibility. Expected 216 from C header, got 192 from PyObject**

**Comment by davidfarinajr** [comment]
@NUSCondy , this looks to me like a cython error.  Have you tried `make` , and if that doesn't work `make clean` and `make` in the rmgpy home folder?

**Comment by NUSCondy** [comment]
Yes I have tried make clean and then make.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by davidfarinajr** [comment]
I see, perhaps @mjohnson541 might know what's going on?

**Comment by mjohnson541** [comment]
That is certainly a cython error, I'm really not sure what to tell you beyond to run make clean and then make again. Did you do the make clean and make in the rmg_env?

**Comment by NUSCondy** [comment]
Oh, I forgot to make in rmgenv. Will update it.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by NUSCondy** [comment]
Hi,
FYI, the RMG runs normally with both the Aramcomech2.0 and Pdep.

**Pull request merged**